.github/scripts/init-pytest-ui.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+set -euo pipefail
+
+echo "UI pre-run: ensuring pytest accepts charm-file argument"
+
+# Create a temporary conftest.py that adds the --charm-file flag
+cat > tests/conftest.py <<'EOF'
+def pytest_addoption(parser):
+    parser.addoption("--charm-file", action="store", default=None)
+EOF
+
+juju add-model testing || juju switch testing || true
+
+if command -v k8s >/dev/null 2>&1; then
+    echo "Enabling ingress in Canonical Kubernetes..."
+    sudo k8s enable ingress || true
+fi
```

Comment:
So we need to enable ingress and create an ingress (nginx) deployment? Does it not work with just the `ingress` from k8s?

Reply:
Closing as per our offline discussion in the daily.